src/parsing/xpath.pegjs

```diff
@@ -530,6 +531,33 @@ quantifiedExprInClause
      ]
    }
 
+// 74 
+TypeswitchExpr
+ = "typeswitch" _ "(" expr:Expr ")" _ clauses:(c:CaseClause _ {return c})+ "default" S varName:("$" v:VarName S {return v})? "return" S predicateExpr:ExprSingle
```

Comment:
predicates are used for filtering, can we just use something like `expr` or even `defaultExpr` for the default return to prevent possible confusion later on.